### DISCUSSION TITLE
KnownTileSources.Create creates an extent for KnownTileSource.StamenTerrain

### DISCUSSION THREAD

**Discussion by Donsroom** [Mapsui/Mapsui#1787] [Q&A] [answered]
I notice that calling KnownTileSources.Create to create a source for StamenTerrain adds an extent that limits the coverage to North America. I worked around this by manually creating the tile source instead of calling KnownTileSources.Create. But is there a reason that the KnownTileSources.Create method creates the extent? Should we not be using Stamen Terrain outside of North America?

Thanks,
Don

**Comment by Donsroom** [comment, marked as answer]
Actually, the KnownTileSources class appears to be part of the BruTile package. So you may disregard my question.

**Reply by pauldendulk** [reply]
At the time when the Stamen known tile source was created it only covered North America, but apparently that is fixed now. The BruTile definition is [here](https://github.com/BruTile/BruTile/blob/726fa1f389d60b9c63bc643d86a4e17a53b39ba7/BruTile/Predefined/KnownTileSources.cs#L118-L126).

What you are saying is that just the extent initialization needs to be removed:

```csharp
{
    Extent = new Extent(-14871588.04, 2196494.41775, -5831227.94199995, 10033429.95725)
},
```

**Reply by Donsroom** [reply]
Thanks so much for the reply. And yes, removing that line gives Stamen Terrain coverage for the entire world.

**Reply by pauldendulk** [reply]
Fixed it in BruTile https://github.com/BruTile/BruTile/pull/196. Also had to set the max zoom level to 0 instead of 4.
